include/annotated.hpp

```diff
@@ -0,0 +1,189 @@
+#ifndef GRAPH_PROTOTYPE_ANNOTATED_HPP
+#define GRAPH_PROTOTYPE_ANNOTATED_HPP
+
+#include <string_view>
+#include <type_traits>
+#include <utils.hpp>
+
+namespace fair::graph {
+
+/**
+ * @brief a template wrapping structure, which holds a static documentation (e.g. mark down) string as its value.
+ * It's used as a trait class to annotate other template classes (e.g. blocks or fields).
+ */
+template<fair::meta::fixed_string doc_string>
+struct Doc {
+    static constexpr fair::meta::fixed_string value = doc_string;
+};
+
+using EmptyDoc = Doc<"">; // nomen-est-omen
+
+template<typename T>
+struct is_doc : std::false_type {};
+
+template<fair::meta::fixed_string N>
+struct is_doc<Doc<N>> : std::true_type {};
+
+template<typename T>
+concept Documentation = is_doc<T>::value;
+
+/**
+ * @brief Unit is a template structure, which holds a static physical-unit (i.e. SI unit) string as its value.
+ * It's used as a trait class to annotate other template classes (e.g. blocks or fields).
+ */
+template<fair::meta::fixed_string doc_string>
+struct Unit {
+    static constexpr fair::meta::fixed_string value = doc_string;
+};
+
+using EmptyUnit = Unit<"">; // nomen-est-omen
+
+template<typename T>
+struct is_unit : std::false_type {};
+
+template<fair::meta::fixed_string N>
+struct is_unit<Unit<N>> : std::true_type {};
+
+template<typename T>
+concept UnitType = is_unit<T>::value;
```

Comment:
Thanks for pointing out the naming issue.  We already have `static_assert(fair::meta::is_instantiation_of<SupportedTypes<float, double>, SupportedTypes>);`. 

Will change and base the check on that... it's a cleaner solution and foregoes the is_... interfaces.

The issue with lengthening `Doc<T>` to `Documentation<T>` is that we want to use this also for fields which would blow-up the line-length a bit ... I would leave it for now since this can be exchanged at a later stage but will put this up for discussion also for the other GR developers.

Reply:
What blows up the line-length is not pressing the return key :stuck_out_tongue_winking_eye:. This is fine:
```c++
struct TestBlock : public node<TestBlock<T>, BlockingIO,
                               Documentation<"some test doc documentation">,
                               SupportedTypes<float, double>> 
```
Once you start using `R""(...)""` you want an alias anyway, no?

It's simpler if users don't have to remember which names to abbreviate and which not. Just type (and auto-complete) full names.